src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.WriteHelpers.cs

```diff
@@ -179,6 +178,7 @@ rootValue is not null &&
                             }
                             else
                             {
+                                writer.Flush();
```

Comment:
Is this fixing an unrelated bug related to `SuppressFlush`?

Reply:
Nope, `writer.Flush()` calls `Advance` on the underlying `PipeWriter` which is what we're trying to avoid here. So without the move, `[` would be written to the `PipeWriter` still before observing the exception from the `IAsyncEnumerable`.